rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,159 @@
+
+- **Element-first (strict element sizing):** In this type the size takes the priority. For examples *buttons*, *form inputs*, *etc*.
+![Element-First Spacing](./images/spatial/element-first.png)
+
+- **Content first (strict internal padding):** This is to be used when the content is dynamic or less predictable. Therefore, we will enforce the strict internal padding to those components. Simply put, this is the spacing around the elements.
+![Content-First Spacing](./images/spatial/content-first.png)
+
+### **Grids**
+The grid system helps align page elements based on sequenced columns and rows. Basically it helps you arrange your structure into structured manner. Here we'll look into 2D organisation of information.
+
+### **Column Grid**
+As the name suggests, this type of grid will help you organise content into evenly spaced vertical columns. The spacer which divide this grid is called as `gutter`. We'll use a **12 column grid** as it helps divide the area into *halfs*, *thirds*, *fourths*, *sixths*.
+
+Here we'll have 3 types of column grids, these will be used depending on the use case.
+1. **Default:** This has a gutter of `32px` & will be used on the internal dashboard.
+![Default Column-Grid](./images/spatial/col-grid-default.png)
+
+2. **Condensed:** This has a gutter of `16px` & will be used on mobile layouts.
+![Condensed Column-Grid](./images/spatial/col-grid-condensed.png)
+
+3. **Spacious:** This has a gutter of `48px` & will be used for landing/marketing pages.
+![Condensed Column-Grid](./images/spatial/col-grid-spacious.png)
+
+These column grids can also be used internally based on the what component we are building. For example, let's say we are building inline form, then we'll use condensed col-grid for organising elements.
+
+### **Spatial Scale**
+We'll follow a **8pt** linear scale for the elements with the **4pt** half-step for spacing smaller content (icons, etc.) and typography (line-heights).
+
+This is how the scale will look like:
+
+| Alias      | Value |
+|------------|-------|
+| $spacer-0  | 0px   |
+| $spacer-1  | 4px   |
+| $spacer-2  | 8px   |
+| $spacer-3  | 12px  |
+| $spacer-4  | 16px  |
+| $spacer-5  | 20px  |
+| $spacer-6  | 24px  |
+| $spacer-7  | 32px  |
+| $spacer-8  | 40px  |
+| $spacer-9  | 48px  |
+| $spacer-10 | 56px  |
+
+>These are mixed values, it contains multiple of 4 as well as 8.
+
+# Drawbacks/Constraints
+There are no major drawbacks here,
+- We need to make sure that all the spacing between the elements should not be hardcoded. They should use the spatial tokens whenever required.
+- We might find few screens where it is difficult to organise the spacing. For example, iPhone 6 (`375x667`). For such scenarios we don't need to break the pixels, instead we need to keep the margins & paddings consistent. It's ok to have oddly sized layout which keeps the grid consistent. Although, no user will likely see the actual measurements here.
```

Comment:
can you post the image as well of what is right way and wrong way for explanation of this point?